test/python/primitives/containers/test_bit_array.py

```diff
@@ -695,3 +695,214 @@ def test_expectation_values(self):
                 _ = ba.expectation_values("Z")
             with self.assertRaisesRegex(ValueError, "is not diagonal"):
                 _ = ba.expectation_values("X" * ba.num_bits)
+
+    def test_postselection(self):
```

Comment:
This test seems to be somewhat indirect for a unit test. It seems to boil down to making sure that `postselect` works for 0-d bit arrays, in which case I wonder if we couldn't just do something like

```python
# manually construct some bitstrings with num_bits > 16, maybe 21 or something.
arr = BitArray.from_bitstrings(
    [
        "01001"....<maybe 4 or 5 bitstrings>
    ]
)

# now test a handful of cases with post-select, manually.
```

I'd be most concerned about testing:
 - various indexes that straddle or don't straddle byte alignments
 - indices that are out of order
 - at least 1 multi-dim case

Reply:
Added more-focused test cases, postselecting on a multi-dimensional BitArray.

Reply:
thanks